### PR TITLE
remove 'getLatLng' property from PointSymbolizer - references #148

### PR DESCRIPTION
**Pull request by chriszrc**
The error is visible on the official vectorgrid demo pages:

http://leaflet.github.io/Leaflet.VectorGrid/demo-points-icons.html

Just open the console and mouse over any of the markers. The error precludes any mouseover or click events on point features. 

My Diagnosis:
Point Symbolizer instances inherit from marker or circlemarker, but they lack latlng information, and on mouseover or click, leaflet will error on these features if it finds the getLatLng method, since that indicates it should be treated as a real marker instance, but even though the method is there, the value isn't, so leaflet produces an error.

Maybe these instance should have a latlng, I don't know, this was just an easy fix so that I could continue to use vectorgrid protobuf support with points and have mouseover and click events work.

### CODE REVIEW

**Comment by tomchadwin** [comment]
My gut feeling is that there should be a latlng.

**Comment by chriszrc** [comment]
I fully agree, but I read through all the source code, and for the life of me, could not figure out how to add that. As I said, I definitely don't know if this is the right answer, but also wasn't getting any traction in the issues section, so I tried to put forth some good faith effort here-

**Comment by tomchadwin** [comment]
Absolutely - I have the same position with a different issue/PR. I just meant that this is probably fine for your fork, but that we need LatLng fixed here. Not trying to belittle your work one bit.

File: src/Leaflet.VectorGrid.js [hunk 1]
```diff
@@ -77,16 +77,16 @@ L.VectorGrid = L.GridLayer.extend({
 				for (var layerName in vectorTile.layers) {
 					this._dataLayerNames[layerName] = true;
 					var layer = vectorTile.layers[layerName];
-
```
Comment (perliedman) [inline src/Leaflet.VectorGrid.js]: Would be nice if we could back out all these changes which I guess are caused by some auto-setting in you editor - while I think all agree that we should not have trailing whitespace, the change history will be confusing if these changes are done as part of other commits.

File: src/Leaflet.VectorGrid.js [hunk 2]
```diff
@@ -217,6 +217,8 @@ L.VectorGrid = L.GridLayer.extend({
 		switch (feat.type) {
 		case 1:
 			layer = new PointSymbolizer(feat, pxPerExtent);
+			//prevent leaflet from treating these canvas points as real markers
+			layer.getLatLng = null;
```
Comment (perliedman) [inline src/Leaflet.VectorGrid.js]: I think this is a fine solution, at least for now, given how LVG hacks a lot of other internals of Leaflet. :+1:

**Comment by chriszrc** [comment]
@perliedman ok, I'll make those changes-

**Comment by bagage** [comment]
Hello @chriszrc, any news on this? Thanks!

File: .gitignore [hunk 3]
```diff
@@ -2,7 +2,6 @@ node_modules
 vendor
 npm-debug.log
 .gobble-build
-dist
```
[the remark this answers is not in the document]
Comment (chriszrc) [reply .gitignore]: Ok, I put this back in

File: src/Leaflet.VectorGrid.js (hunk 1 above)
Comment (chriszrc) [reply src/Leaflet.VectorGrid.js]: I backed out the whitespace changes

**Comment by chriszrc** [comment]
@perliedman Ok, I think I backed out the changes to dist and whitespace, let me know if there's anything else I can do to get this change merged-

**Comment by perliedman** [comment]
@chriszrc nice! It still looks like the `dist/` files are part of the PR, though - they all show up under "Files changed". Also, looks like some unrelated changes to `package-lock.json` snuck in there, just get rid of that changes as well, and I think we are good to go!

**Comment by chriszrc** [comment]
Ok, I'm looking at this page now:

https://github.com/Leaflet/Leaflet.VectorGrid/pull/157/files

And it seems like maybe I got everything this time-

**Comment by perliedman** [comment]
Nice! Thanks for all the fixes, great!

**Comment by bagage** [comment]
Any chance to publish a new version on npm with this fix? Thanks!

**Comment by SeeYouTomorrow** [comment]
Hi, i came across the mentioned bug recently, and tried to fixed it by replacing leaflet.vectorgrid in package.json with @chriszrc fork, then i tried to run yarn install inside node_modules/leaflet.vectorgrid since i noticed that dist folder is missing, however i was unable to make it work because of errors in runas build (runas is one of dependencies)....Does anyone have similiar experience or can at least point me in right direction plz ? Im on linux btw

some lines from error message that could be of some use : 
- .../node_modules/leaflet.vectorgrid/node_modules/runas: Command failed.
- runas.target.mk:103: recipe for target 'Release/obj.target/runas/src/main.o' failed

**Pull request merged**